cobc/typeck.c

```diff
@@ -1055,16 +1055,31 @@ cb_emit_list (cb_tree l)
 	return l;
 }
 
+static COB_INLINE COB_A_INLINE int
+cb_tree_is_numeric_ref_or_field (cb_tree x) {
+	if (x
+	 && x != cb_error_node
+	 && CB_REF_OR_FIELD_P (x)
+	 && CB_TREE_CATEGORY (x) == CB_CATEGORY_NUMERIC) {
```

Comment:
no need to check for error node - that has no numeric category
so possibly just for x, then get the category - it has to be also generated for `CB_CATEGORY_NUMERIC_EDITED`!

Reply:
So, this changes the previous behavior of `cb_emit_incompat_data_checks`, right?

Reply:
Ignoring `cb_error_node` is fine.
But handling `CB_CATEGORY_NUMERIC_EDITED` in source fields seems to break many tests.
Allowing checks in case numeric-edited receivers makes sense, though: I'll include this (and add that in the test as well).